src/backend/legendary/user.ts

```diff
@@ -12,32 +16,57 @@ export class LegendaryUser {
   public static async login(authorizationCode: string) {
     const commandParts = ['auth', '--code', authorizationCode]
 
-    try {
-      await runLegendaryCommand(commandParts, {
-        logMessagePrefix: 'Logging in'
-      })
-      const userInfo = await this.getUserInfo()
-      return { status: 'done', data: userInfo }
-    } catch (error) {
+    const abortID = 'legendary-login'
+    const errorMessage = (error: string) => {
       logError(['Failed to login with Legendary:', error], {
         prefix: LogPrefix.Legendary
       })
 
       return { status: 'failed' }
     }
+
+    try {
+      const res = await runLegendaryCommand(
+        commandParts,
+        createAbortController(abortID),
+        {
+          logMessagePrefix: 'Logging in'
+        }
+      )
+
+      deleteAbortController(abortID)
+
+      if (res.error || res.abort) {
+        return errorMessage(res.error ?? 'abort by user')
+      }
+
+      const userInfo = await this.getUserInfo()
+      return { status: 'done', data: userInfo }
+    } catch (error) {
+      deleteAbortController(abortID)
+      return errorMessage(`${error}`)
+    }
   }
 
   public static async logout() {
     const commandParts = ['auth', '--delete']
 
-    const res = await runLegendaryCommand(commandParts, {
-      logMessagePrefix: 'Logging out'
-    })
+    const abortID = 'legendary-logout'
+    const res = await runLegendaryCommand(
+      commandParts,
+      createAbortController(abortID),
```

Comment:
I wonder if we can have the abort controller as an optional argument in the options hash instead of a fixed second argument, I see many places where we call `runLegendaryCommand` and `runGogCommand` that I understand is just adding the abort controller and deleting this after but we won't really abort a login for example, should make things cleaner, like... only pass an abort controller if we want to but not have to do it always

Reply:
I second this. Having to always pass an AbortController is adding a lot of unnecessary code.

I get that for *some actions*, you want to have it so it can be aborted from somewhere else, but it's a little excessive to always provide it.
In this instance (logging out), the process will only live for at most half second, I doubt anything will try to cancel it (and besides, how would that even happen? A "Cancel logging out" button seems very strange to me)

Reply:
I agree but I think it is out of the scope of this PR.

Reply:
I like to have everywhere a abortcontroller, because if someone closes heroic during login and legendary or gogdl stuck, they will not be closed. Because of the abortcontroller on closing all running child_processes will be aborted.

Reply:
Maybe what we can do is make this argument optional and, if the option is not passed, the `runXYZCommand`  methods can create an abort controller internally.

That way we can have an explicit abort controller when needed if we wan to be able to abort on demand, and fallback to an internally-managed abortController to handle that case of something that could get stuck without having to always pass it.

Also, would it make sense to automatically remove the abort controller right before returning from those methods? so we don't have to do the deleteAbortController everywhere, if the methods get to the last line to return a value it means the abort controller can be deleted so we could hide that code inside the `runXYZCommand` methods.